esp/services/ws_dfu/ws_dfuService.cpp

```diff
@@ -6507,8 +6508,14 @@ bool CWsDfuEx::onDFUFilePublish(IEspContext &context, IEspDFUFilePublishRequest
                 throw makeStringExceptionV(ECLWATCH_FILE_NOT_EXIST, "DFUFilePublish: Failed to find group %s.", groupName);
 
             fileDesc.setown(createFileDescriptor(normalizeTempFileName, clusterTypeEx, groupName, group));
-            if (compressed)
-                fileDesc->queryProperties().setPropBool("@blockCompressed", true);
+            if (fileIdItems.ordinality()>3) // compressed
+            {
+                bool compressed = strToBool(fileIdItems.item(3));
+                if (compressed)
+                    fileDesc->queryProperties().setPropBool("@blockCompressed", true);
+            }
+            if (fileIdItems.ordinality()>4) // fileType
+                fileDesc->queryProperties().setProp("@kind", fileIdItems.item(4));
```

Comment:
Should the logic here be: if fileIdItems.ordinality()>4, read both compressed and fileType, else if fileIdItems.ordinality()>3, read the fileType only?

Reply:
well, I should move this inside the >3 block, as impossible if not >3

Reply:
@wangkx - have move per above.

Reply:
@jakesmith if (req.getCompressed()) returns false, will the third item of the fileIdItems be fileType?

Reply:
@wangkx - hm, well spotted.
It would be invalid/wrong if getCompressed() were false.

Will fix by adding the compressed field unconditionally.

Reply:
@wangkx - have fixed